src/main/java/org/candlepin/service/EventAdapter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2009 - 2024 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service;
+
+import org.candlepin.service.model.Event;
+
+/**
+ * An interface for publishing Candlepin {@link Event}s.
+ */
+public interface EventAdapter {
+
+    /**
+     * An application lifecycle method called during the initialization sequence.
+     */
+    void initialize();
+
+    /**
+     * An application lifecycle method called during the shutdown sequence.
+     */
+    void shutdown();
+
+    /**
+     * Publishes the provided event.
```

Comment:
What should implementers do if event is null? We need to describe more of the intended behavior for all cases, not just the general case.

Reply:
Updated to hopefully make the interface more clear